apis/core/lib/auth.ts

```diff
@@ -83,7 +86,38 @@ export default async () => {
     router.use(devAuthHandler)
   }
 
-  router.use(setUserId)
+  router.use(setUserId).use(updateUserResource())
 
   return router
 }
+
+function updateUserResource(): RequestHandler {
+  const users = new TermSet()
+
+  return (req, res, next) => {
+    if (req.user?.id && !users.has(req.user.id)) {
+      const { id, name, sub } = req.user
+      users.add(id)
+
+      req.once('end', () => {
+        DELETE`
+          GRAPH ${id} {
+            ${id} ${schema.name} ?name .
```

Comment:
If users live in their own named graph, wouldn't it be cleaner to delete the graph as a whole?

Reply:
Well, this is but not deleting the whole graph. Only `schema:name` :)